packages/core/src/input/InputManager.ts

```diff
@@ -0,0 +1,346 @@
+import { Ray, Vector2 } from "@oasis-engine/math";
+import { Engine } from "../Engine";
+import { Entity } from "../Entity";
+import { HitResult } from "../HitResult";
+import { PhysicsManager } from "../PhysicsManager";
+import { SceneManager } from "../SceneManager";
+import { Script } from "../Script";
+import { Input } from "./Input";
+import { Pointer } from "./Pointer";
+
+/** @internal */
+enum PointerEventType {
+  PointDown,
+  PointUp,
+  PointMove
+}
+
+/** @internal */
+enum PointerChangeType {
+  Update,
+  Remove
+}
+
+/**
+ * Input Manager.
+ */
+export class InputManager {
+  /** Canvas to listen for input. */
+  private _canvas: HTMLCanvasElement;
+  /** SceneManager. */
+  private _sceneMgr: SceneManager;
+  /** PhysicsManager. */
+  private _physicsMgr: PhysicsManager;
```

Comment:
Cache `engine` instead of `_sceneMgr` and `_physicsMgr ` is enough!

Reply:
Done.